### PR TITLE
Ensure that container image tests only run on linux when the docker flag is enabled

### LINKED ISSUE #7585: quarkus master build fails if no docker daemon is running
With the addition of the container image for Jib and friends, the quarkus build no longer run if no docker daemon is present.
These tests should only be run if the docker profile is in use.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #7585

### CODE REVIEW

File: integration-tests/container-image/pom.xml [hunk 1]
```diff
@@ -9,6 +9,9 @@
   </parent>
   <artifactId>quarkus-integration-test-container-image</artifactId>
   <name>Quarkus - Integration Tests - Jib extension</name>
```
Comment (gsmet) [inline integration-tests/container-image/pom.xml]: The name looks a bit off?

**Comment by gsmet** [comment]
@geoand I promoted it to get an earlier CI result as I would like to have that one in CR1.

File: integration-tests/container-image/pom.xml (hunk 1 above)
Comment (geoand) [reply integration-tests/container-image/pom.xml]: Indeed, thanks for spotting it! It should be fixed now

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Restarted CI as we had MongoDB failing to start and all the JVM tests got cancelled :/

**Comment by geoand** [comment]
Damn mongo...

**Comment by geoand** [comment]
We need to get rid of that embedded mongo thing. I'll try and take a look soon

**Comment by machi1990** [comment]
> We need to get rid of that embedded mongo thing. I'll try and take a look soon

+1 We can use a mongo container instead? or testcontainers?

**Comment by gsmet** [comment]
@machi1990 yeah whatever is more stable than what we have. I would go with a Docker container started from the pom as for the other tests if doable.

**Comment by geoand** [comment]
That's my idea yes. The only issue I see now is that currently the tests don't need `docker` to run, meaning that you can run them on any machine (which is desirable for the tests that are part of the extensions themselves)

**Comment by geoand** [comment]
But at this point I also believe that having more stability is more desirable than being able to run all the extension tests without docker

**Comment by geoand** [comment]
Took a quick look. We can get rid of most of the `de.flapdoodle.embed.mongo` tests, but the ones that use a replica set I don't think we can...

**Pull request merged**